ENIGMAsystem/SHELL/Platforms/Cocoa/CocoaMain.cpp

```diff
@@ -32,32 +32,63 @@
 #include "../General/PFfilemanip.h"
 
 #include "Platforms/General/PFmain.h"
+#include "Platforms/General/PFfilemanip.h"
 #include "Universal_System/roomsystem.h"
+#include "Universal_System/estring.h"
+
+/* Uncomment the 2 lines below for testing generate_working_directory() success/failure */
+// #include "Widget_Systems/widgets_mandatory.h"
+// #include "Widget_Systems/Cocoa/dialogs.h"
 
 using std::string;
 
+using enigma_user::set_working_directory;
+using enigma_user::directory_exists;
+using enigma_user::filename_name;
+using enigma_user::filename_dir;
+using enigma_user::filename_path;
+using enigma_user::filename_ext;
+
 namespace enigma {
 
 void SetResizeFptr();
 
+static inline void generate_working_directory() {
+  /* This function will set the working directory to the app bundle's Resources folder 
+  like GM4Mac 7.5, GMStudio 1.4, GMS 2.x and most Mac apps do, if the executable is in
+  an app bundle. If the executable is not in an app bundle, use the getcwd function */
+  bool success = false; string macos_bname = filename_name(filename_dir(enigma_user::program_directory));
```

Comment:
This is very difficult to read, in this format.

Reply:
I agree and disagree. I would extract the inner `filename_dir` to another variable though. I would also probably break this into two lines or more. Finally, use a comma to separate statements on the same line, not a semicolon.

```cpp
bool success = false;
string macos_dname = filename_dir(enigma_user::program_directory),
          macos_bname = filename_name(macos_dname);
```

Reply:
Also, I don't like when multi-line comments begin on the same line as the opening comment.